contracts/stark-proxy/v2/impl/SP2Exchange.sol

```diff
@@ -0,0 +1,275 @@
+
+  // ============ Constants ============
+
+  IStarkPerpetual public immutable STARK_PERPETUAL;
+
+  // ============ Events ============
+
+  event DepositedToExchange(
+    uint256 starkKey,
+    uint256 starkAssetType,
+    uint256 starkVaultId,
+    uint256 tokenAmount
+  );
+
+  event WithdrewFromExchange(
+    uint256 starkKey,
+    uint256 starkAssetType,
+    uint256 tokenAmount,
+    bool isGuardianAction
+  );
+
+  /// @dev Limited fields included. Details can be retrieved from Starkware logs if needed.
+  event RequestedForcedWithdrawal(
+    uint256 starkKey,
+    uint256 vaultId,
+    bool isGuardianAction
+  );
+
+  /// @dev Limited fields included. Details can be retrieved from Starkware logs if needed.
+  event RequestedForcedTrade(
+    uint256 starkKey,
+    uint256 vaultId,
+    bool isGuardianAction
+  );
+
+  /// @dev Limited fields included. Details can be retrieved from Starkware logs if needed.
+  event DepositCanceled(
+    uint256 starkKey,
+    uint256 vaultId,
+    bool isGuardianAction
+  );
+
+  /// @dev Limited fields included. Details can be retrieved from Starkware logs if needed.
+  event DepositReclaimed(
+    uint256 starkKey,
+    uint256 vaultId,
+    bool isGuardianAction
+  );
+
+  // ============ Constructor ============
+
+  constructor(
+    IStarkPerpetual starkPerpetual
+  ) {
+    STARK_PERPETUAL = starkPerpetual;
+  }
+
+  // ============ External Functions ============
+
+  /**
+   * @notice Deposit funds to the exchange.
+   *
+   *  IMPORTANT: The caller is responsible for providing `quantizedAmount` in the right units.
+   *             Currently, the exchange collateral is USDC, denominated in ERC20 token units, but
+   *             this could change.
+   *
+   * @param  starkKey         The STARK key of the account. Must be authorized by OWNER_ROLE.
+   * @param  assetType        The exchange asset ID for the asset to deposit.
+   * @param  vaultId          The exchange position ID for the account to deposit to.
+   * @param  quantizedAmount  The deposit amount denominated in the exchange base units.
+   *
+   * @return The ERC20 token amount spent.
+   */
+  function depositToExchange(
+    uint256 starkKey,
+    uint256 assetType,
+    uint256 vaultId,
+    uint256 quantizedAmount
+  )
+    external
+    nonReentrant
+    onlyRole(EXCHANGE_OPERATOR_ROLE)
+    onlyAllowedKey(starkKey)
+    returns (uint256)
+  {
+    // Deposit and get the deposited token amount.
+    uint256 startingBalance = getTokenBalance();
+    STARK_PERPETUAL.deposit(starkKey, assetType, vaultId, quantizedAmount);
+    uint256 endingBalance = getTokenBalance();
+    uint256 tokenAmount = startingBalance.sub(endingBalance);
+
+    // Disallow depositing borrowed funds to the exchange if the guardian has restricted borrowing.
+    if (_IS_BORROWING_RESTRICTED_) {
+      require(
+        endingBalance >= getBorrowedAndDebtBalance(),
+        'SP2Exchange: Cannot deposit borrowed funds to the exchange while Restricted'
```

Comment:
This error string previously had `SP1Borrowing`, and I changed it to `SP2Exchange`. Why was it originally `SP1Borrowing`?

Reply:
Was a mistake